Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -6981,14 +6981,21 @@ static RefPtr<CSSPathValue> consumeBasicShapePath(CSSParserTokenRange& args, Opt
             return nullptr;
     }
 
-    if (args.peek().type() != StringToken)
-        return nullptr;
+    auto pathValueFromStringView = [&](const StringView& view) -> RefPtr<CSSPathValue> {
+        SVGPathByteStream byteStream;
+        if (!buildSVGPathByteStreamFromString(view, byteStream, UnalteredParsing) || byteStream.isEmpty())
+            return nullptr;
+        return CSSPathValue::create(WTFMove(byteStream), rule);
+    };
 
-    SVGPathByteStream byteStream;
-    if (!buildSVGPathByteStreamFromString(args.consumeIncludingWhitespace().value(), byteStream, UnalteredParsing) || byteStream.isEmpty())
+    switch (args.peek().type()) {
+    case IdentToken:
+        return pathValueFromStringView(args.consumeAll().serialize());
+    case StringToken:
+        return pathValueFromStringView(args.consumeIncludingWhitespace().value());
```

Comment:
Should we verify that they are no token pending ? 

```
if (!args.atEnd())
      return nullptr;
```

Reply:
The call sites do. I'm not sure whether we should factor this into `consumeBasicShapePath()` or let the call sites handle it. See for instance `consumeBasicShape()` which has this single check at its end:

```
    if (!result || !args.atEnd())
        return nullptr;
```

Reply:
This code has been removed.